lib/ash_authentication/strategies/api_key/plug.ex

```diff
@@ -138,8 +138,10 @@ defmodule AshAuthentication.Strategy.ApiKey.Plug do
   defp get_api_key(conn, %{source: :header, header_prefix: prefix}) do
     case Conn.get_req_header(conn, "authorization") do
       [header_value | _] ->
-        if String.starts_with?(header_value, prefix) do
-          {:ok, String.replace_prefix(header_value, prefix, "")}
+        prefix = Regex.compile!("^" <> prefix, "i")
```

Comment:
🤔 this is actually an interesting question, I'm not sure if this should be case insensitive by default or not, but perhaps instead we allow the prefix to be a regex? That would allow this behavior without making it the default.

Reply:
Fair enough! Looks like it's supposed to be case sensitive: https://datatracker.ietf.org/doc/html/rfc6750#section-1.1. I mostly added this because I was wondering why my requests were erroring as unauthorized.

I like the idea of passing in a regex though. Happy to change this.